### PR TITLE
modules/wayland: Support --include-core-only

### PR DESCRIPTION
**Pull request by markbolhuis**
wayland-scanner can generate header files that only include
wayland-client-core.h using a flag.

Add a core_only option to scan_xml to support this use case.

### CODE REVIEW

File: docs/markdown/Wayland-module.md [hunk 1]
```diff
@@ -61,6 +62,8 @@ It takes the following keyword arguments:
   generated. The default is true.
 - `server` Optional arg that specifies if server side header file is
   generated. The default is false.
+- `core_only` Optional arg the specifies that generated headers only include
```
Comment (xclaesse) [inline docs/markdown/Wayland-module.md:65]:
```suggestion
- `core_only` Optional arg that specifies that generated headers only include
```

**Review by xclaesse — COMMENTED** [review]
LGTM

File: mesonbuild/modules/wayland.py [hunk 2]
```diff
@@ -98,12 +100,16 @@ def scan_xml(self, state: ModuleState, args: T.Tuple[T.List[FileOrString]], kwar
             targets.append(code)
 
             for side in sides:
+                command = [self.scanner_bin, f'{side}-header', '@INPUT@', '@OUTPUT@']
+                if kwargs['core_only']:
+                    command.append('-c')
```
Comment (tristan957) [inline mesonbuild/modules/wayland.py:105]: I think you should use the long argument instead of the short one. Makes reading the code a little easier.

Comment (xclaesse) [reply mesonbuild/modules/wayland.py:105]: Oh good point, did not notice there is `--include-core-only`.

**Comment by tristan957** [comment]
@xclaesse good to merge?

**Comment by tristan957** [comment]
Giving this some more thought, should the option be named `include_core_only` to match the long option name?

**Comment by markbolhuis** [comment]
> Giving this some more thought, should the option be named `include_core_only` to match the long option name?

I think it's fine, the option cannot be mistaken for anything else.

Side note: The inclusion of `wayland-<client|server>.h` is discouraged, so it might be worth considering in the future if this option should be opt out rather than opt in, or perhaps the option is removed and headers are always generated with `--include-core-only`

**Comment by tristan957** [comment]
I think a PR to change the default to true makes sense. Has this always been the case?

**Comment by markbolhuis** [comment]
`wayland-<client|server>-core.h` was added [in 2015](https://gitlab.freedesktop.org/wayland/wayland/-/commit/d74a9c079b1aeb44f69b4132dc2c38362e21f281).

**Comment by tristan957** [comment]
Then seems like changing the default would be for the best. Meson wants to promote best practices with its modules.

> I think it's fine, the option cannot be mistaken for anything else.

Probably, but Meson tends to be more explicit when possible.

**Comment by xclaesse** [comment]
The module is still unstable, so it's fine to change the default if it's recommended to only include core by default. Please open a new PR, we went too fast here it seems.

**Comment by markbolhuis** [comment]
Will do

**Pull request merged**